Formula/c/cloud-provider-kind.rb

```diff
@@ -0,0 +1,20 @@
+class CloudProviderKind < Formula
+  desc "Cloud provider for KIND clusters"
+  homepage "https://github.com/kubernetes-sigs/cloud-provider-kind"
+  url "https://github.com/kubernetes-sigs/cloud-provider-kind/archive/refs/tags/v0.5.0.tar.gz"
+  sha256 "ba3fad794370b1ff14e608edfa5883cb455e1e6e89d6a63187c5f47bf4e23251"
+  license "Apache-2.0"
+  head "https://github.com/kubernetes-sigs/cloud-provider-kind.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+  end
+
+  test do
+    # Verify that the helper is running properly.
+    status_output = shell_output("#{bin}/cloud-provider-kind -h 2>&1")
+    assert_match "Usage: cloud-provider-kind", status_output
```

Comment:
need a better test than help or version check.

Reply:
what about something that asserts on the absence of docker or podman
```
/cloud-provider-kind
I0202 17:47:43.919164   61883 app.go:65] FLAG: --enable-lb-port-mapping="false"
I0202 17:47:43.919184   61883 app.go:65] FLAG: --enable-log-dumping="false"
I0202 17:47:43.919187   61883 app.go:65] FLAG: --logs-dir=""
I0202 17:47:43.919190   61883 app.go:65] FLAG: --v="2"
F0202 17:47:43.919511   61883 app.go:154] can not detect cluster provider: failed to detect any supported node provider

```

it similar to the one in `kind` https://github.com/Homebrew/homebrew-core/blob/bfb2a310a8767bc0ff97ff560b0d0ab6e5e40f97/Formula/k/kind.rb#L28-L32

Reply:
I changed to this.

```
    ENV["DOCKER_HOST"] = "unix://#{testpath}/invalid.sock"
    status_output = shell_output("#{bin}/cloud-provider-kind 2>&1", 255)
    if OS.mac?
      # Should error out as requires root on Mac
      assert_match "Please run this again with `sudo`", status_output
    elsif OS.linux?
      # Should error out because without docker or podman
      assert_match "can not detect cluster provider", status_output
    end
```

What do you think?